src/main/java/jnr/unixsocket/UnixSocketChannel.java

```diff
@@ -18,243 +18,299 @@
 
 package jnr.unixsocket;
 
+import java.io.IOException;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.UnsupportedAddressTypeException;
+
 import jnr.constants.platform.Errno;
 import jnr.constants.platform.ProtocolFamily;
 import jnr.constants.platform.Sock;
 import jnr.constants.platform.SocketLevel;
 import jnr.constants.platform.SocketOption;
-import jnr.enxio.channels.NativeSocketChannel;
-import jnr.ffi.*;
+import jnr.enxio.channels.AbstractNativeSocketChannel;
+import jnr.ffi.LastError;
 import jnr.ffi.byref.IntByReference;
 
-import java.io.IOException;
-import java.nio.channels.SelectionKey;
-import java.util.concurrent.locks.ReadWriteLock;
-import java.util.concurrent.locks.ReentrantReadWriteLock;
-
 /**
- * A {@link java.nio.channels.Channel} implementation that uses a native unix socket
+ * A {@link java.nio.channels.Channel} implementation that uses a native unix
+ * socket
```

Comment:
Try to avoid formatting-only changes. The tabs throughout this PR and 80-col wrapping changes are clouding the PR changes. I will continue to review with whitespace turned off.